packages/jni-swig-stub/realm.i

```diff
@@ -88,6 +88,22 @@ std::string rlm_stdstr(realm_string_t val)
     };
 }
 
+// This sets up a type map for all methods with a (partial) argument pattern of:
+//    realm_synchronous_callback_func_t, void* userdata
+// This will make Swig wrap methods taking this argument pattern into:
+//  - a Java method that takes one argument of type `Object` (`jstype`) and passes this object on as `Object` to the native method (`jtype`+``javain`)
+//  - a JNI method that takes a `jobject` (`jni`) that translates the incoming single argument into the actual three arguments of the C-API method (`in`)
+%typemap(jstype) (realm_should_compact_on_launch_func_t, void* userdata) "Object" ;
+%typemap(jtype) (realm_should_compact_on_launch_func_t, void* userdata) "Object" ;
+%typemap(javain) (realm_should_compact_on_launch_func_t, void* userdata) "$javainput";
+%typemap(jni) (realm_should_compact_on_launch_func_t, void* userdata) "jobject";
+%typemap(in) (realm_should_compact_on_launch_func_t, void* userdata) {
+    auto jenv = get_env(true);
+    $1 = reinterpret_cast<realm_should_compact_on_launch_func_t>(realm_should_compact_callback);
+    // FIXME How to release this: https://github.com/realm/realm-core/issues/5222
```

Comment:
From the comment in the core issue, it seems that we should just find a way to ensure that the configuration is kept a live across `realm_open` and then we don't have to keep the global reference. Swig has some feature to keep an argument across the native call in which case we can maybe achieve this by passing the Kotlin configuration down to the interop layer. Don't know how to achieve in native though 🤔

Reply:
I don't think that is enough? 🤔 A configuration object lives independently from realm_open and can e.g. be used to open multiple Realms.

Reply:
But if the configuration itself is referenced to open other realms, etc.,  then all the callbacks are also kept alive and we wouldn't need the global/stable ref at the core level to keep it alive. But I guess it depends on the context that the callback is actually triggered in.

Reply:
The problem is `jobject` on the JNI side, they are only valid for the duration of the JNI call, regardless of the object in JVM being GC'ed or not. So when we set the callback in the configuration we need to create a global reference to the callback on the JVM side, otherwise we get `invalid object` when opening the Realm, since that happens in another JNI call.